src/Client.php

```diff
@@ -142,7 +142,7 @@ private function buildUri($uri, array $config)
             return $uri instanceof UriInterface ? $uri : new Psr7\Uri($uri);
         }
 
-        return Psr7\Uri::resolve(Psr7\uri_for($config['base_uri']), $uri);
+        return Psr7\Uri::resolve($config['base_uri'], $uri);
```

Comment:
I need to revert that I just realize because the base_uri can be overwritten dynamically through the options in the `request` method. So it's not guaranteed that the base_uri is still a UriInterface instance. I'll add a test for this.